### PR TITLE
[TASK-1] Add avg rpc time for each operation

### PR DESCRIPTION
**Pull request by zhenjiaguo**
### _Why are the changes needed?_

Subtask of #1393. Add avg rpc time for each operation on Kyuubi server.

Use MetricRegistry default histogram to record.

Screenshot of the json file generated by the json reporter:

Screenshot from 2021-12-11 09-20-35

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/DataGenerator.scala [hunk 1]
```diff
@@ -570,31 +570,31 @@ object DataGenerator {
...
-     item,
-     income_band,
-     web_site,
-     web_page,
-     date_dim,
-     time_dim,
-     ship_mode,
-     reason,
-     promotion)
+    Seq(
```
Comment (zhenjiaguo) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/DataGenerator.scala:573]: I use `dev/reformat` to reformat the code. This file format is no correct. Ignore this file when review.

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala [hunk 2]
```diff
@@ -40,4 +40,5 @@ object MetricsConstants {
   final val OPERATION_FAIL: String = OPERATION + "failed"
   final val OPERATION_TOTAL: String = OPERATION + "total"
 
+  final val OPERATION_RPC_TIME: String = OPERATION + "rpc_time"
```
Comment (yaooqinn) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConstants.scala:43]: how about rpc_ms or rpc_millis?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      timeCost += System.currentTimeMillis() - startTime
+    }
+  }
+
+  override def openSession(
+      protocol: TProtocolVersion,
+      user: String,
+      password: String,
+      configs: Map[String, String]): SessionHandle = {
+    time(client.openSession(protocol, user, password, configs))
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:53]: this is for session RPC, why do we wrap them to track as operation metrics. IIUC, they are not used acturally.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  override def closeSession(): Unit = {
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:56]: ditto

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  /**
+   * Get the time cost millis on each call of the function in the [[KyuubiThriftClient]]
+   */
+  def getTimeCostMillis: Long = {
+    timeCost
+  }
+
+  @throws[Exception]
+  private def time[T](f: => T): T = {
+    val startTime = System.currentTimeMillis()
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:40]: shall we use `System.nanoTime` ?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  override def closeSession(): Unit = {
+    client.closeSession()
+  }
+
+  override def executeStatement(
+      statement: String,
+      shouldRunAsync: Boolean,
+      queryTimeout: Long): TOperationHandle = {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:63]: time {
  xxx
}

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    time(client.executeStatement(statement, shouldRunAsync, queryTimeout))
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:64]:
```scala
time {
  xxx
}
```

Comment (ulysses-you) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:64]: all methods should use this style

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala (hunk 5 above)
Comment (zhenjiaguo) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala:40]: I think it is no need to use such precision.

**Pull request closed without merging**